src/Editor/ImGuizmo/imguizmo.cpp

```diff
@@ -0,0 +1,1719 @@
+    {
+        return (v1.x * v2.x) + (v1.y * v2.y) + (v1.z * v2.z);
+    }
+
+    vec_t BuildPlan(const vec_t & p_point1, const vec_t & p_normal)
+    {
+        vec_t normal, res;
+        normal.Normalize(p_normal);
+        res.w = normal.Dot(p_point1);
+        res.x = normal.x;
+        res.y = normal.y;
+        res.z = normal.z;
+        return res;
+    }
+
+    struct matrix_t
+    {
+    public:
+
+        union
+        {
+            float m[4][4];
+            float m16[16];
+            struct
+            {
+                vec_t right, up, dir, position;
+            } v;
+            vec_t component[4];
+        };
+
+        matrix_t(const matrix_t& other) { memcpy(&m16[0], &other.m16[0], sizeof(float) * 16); }
+        matrix_t() {}
+
+        operator float * () { return m16; }
+        operator const float* () const { return m16; }
+        void Translation(float _x, float _y, float _z) { this->Translation(makeVect(_x, _y, _z)); }
+
+        void Translation(const vec_t& vt)
+        {
+            v.right.Set(1.f, 0.f, 0.f, 0.f);
+            v.up.Set(0.f, 1.f, 0.f, 0.f);
+            v.dir.Set(0.f, 0.f, 1.f, 0.f);
+            v.position.Set(vt.x, vt.y, vt.z, 1.f);
+        }
+
+        void Scale(float _x, float _y, float _z)
+        {
+            v.right.Set(_x, 0.f, 0.f, 0.f);
+            v.up.Set(0.f, _y, 0.f, 0.f);
+            v.dir.Set(0.f, 0.f, _z, 0.f);
+            v.position.Set(0.f, 0.f, 0.f, 1.f);
+        }
+        void Scale(const vec_t& s) { Scale(s.x, s.y, s.z); }
+
+        matrix_t& operator *= (const matrix_t& mat)
+        {
+            matrix_t tmpMat;
+            tmpMat = *this;
+            tmpMat.Multiply(mat);
+            *this = tmpMat;
+            return *this;
+        }
+        matrix_t operator * (const matrix_t& mat) const
+        {
+            matrix_t matT;
+            matT.Multiply(*this, mat);
+            return matT;
+        }
+
+        void Multiply(const matrix_t &matrix)
+        {
+            matrix_t tmp;
+            tmp = *this;
+
+            FPU_MatrixF_x_MatrixF((float*)&tmp, (float*)&matrix, (float*)this);
+        }
+
+        void Multiply(const matrix_t &m1, const matrix_t &m2)
+        {
+            FPU_MatrixF_x_MatrixF((float*)&m1, (float*)&m2, (float*)this);
+        }
+
+        float GetDeterminant() const
+        {
+            return m[0][0] * m[1][1] * m[2][2] + m[0][1] * m[1][2] * m[2][0] + m[0][2] * m[1][0] * m[2][1] -
+                m[0][2] * m[1][1] * m[2][0] - m[0][1] * m[1][0] * m[2][2] - m[0][0] * m[1][2] * m[2][1];
+        }
+
+        float Inverse(const matrix_t &srcMatrix, bool affine = false);
+        void SetToIdentity()
+        {
+            v.right.Set(1.f, 0.f, 0.f, 0.f);
+            v.up.Set(0.f, 1.f, 0.f, 0.f);
+            v.dir.Set(0.f, 0.f, 1.f, 0.f);
+            v.position.Set(0.f, 0.f, 0.f, 1.f);
+        }
+        void Transpose()
+        {
+            matrix_t tmpm;
+            for (int l = 0; l < 4; l++)
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') identifier should not be composed of only 'l' and 'O' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT010)